### PR TITLE
Mark scale as const and remove --fp8 flag usage 

### PR DESCRIPTION
**Pull request by Yantom1**
* Mark only scales as const

* remove --fp8 flag usage from llama

* removed usage of ENABLE_CONST_MARKING

Change-Id: I6dba8691d842fc62d09da5202ea1e61a111f5f18

---------

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by regisss** [comment]
Is this linked to #966 ?

**Comment by Yantom1** [comment]
[regisss](https://github.com/regisss) yes its connected

**Review by ssarkar2 — COMMENTED** [review]
1. Can you put the changes of https://github.com/huggingface/optimum-habana/pull/966 into this PR, then it will be easy to merge together, because both are related.

2. Can this be run on 1.15 docker/driver? If so we will test it and merge it fast since its not 1.16 dependent

3. Can you modify this test, because your chaneg removes fp8: https://github.com/huggingface/optimum-habana/blob/3a14236643916ecc5112f23560874cca083f2044/tests/test_text_generation_example.py
esp these lines: https://github.com/huggingface/optimum-habana/blob/3a14236643916ecc5112f23560874cca083f2044/tests/test_text_generation_example.py#L182

**Comment by Yantom1** [comment]
> 1. Can you put the changes of [Remove --fp8 flag from README #966](https://github.com/huggingface/optimum-habana/pull/966) into this PR, then it will be easy to merge together, because both are related.
> 2. Can you modify this test, because your chaneg removes fp8: https://github.com/huggingface/optimum-habana/blob/3a14236643916ecc5112f23560874cca083f2044/tests/test_text_generation_example.py
>    esp these lines: https://github.com/huggingface/optimum-habana/blob/3a14236643916ecc5112f23560874cca083f2044/tests/test_text_generation_example.py#L182

Hey, both fixed.

**Comment by regisss** [comment]
@Yantom1 It seems there are some merge conflicts, can you merge main into your branch and fix them please? Let me know if there is anything you're not sure about.

**Comment by Yantom1** [comment]
> @Yantom1 It seems there are some merge conflicts, can you merge main into your branch and fix them please? Let me know if there is anything you're not sure about.

@regisss done

**Review by ssarkar2 — APPROVED** [review]

File: examples/text-generation/README.md [hunk 1]
```diff
@@ -361,22 +357,6 @@ QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
...
-QUANT_CONFIG=./quantization_config/maxabs_measure.json python run_lm_eval.py \
--o acc_phi-2_bs1_measure.txt  \
---model_name_or_path microsoft/phi-2 \
---use_hpu_graphs \
---use_kv_cache \
---max_new_tokens 100 \
---batch_size 1 \
---trim_logits \
---reuse_cache \
---bf16
```
Comment (regisss) [inline examples/text-generation/README.md:379]: Why removing the measure command for phi but not for other models?

Comment (Yantom1) [reply examples/text-generation/README.md:379]: It's not removed, I put it back in the merge commit afterwards

Comment (regisss) [reply examples/text-generation/README.md:379]: I still see it removed: https://github.com/huggingface/optimum-habana/pull/962/files#diff-d8103ff4973ad399e0cc6fc01efe091f716c7b91510ab34a05ea1d70f180459bL367

Comment (Yantom1) [reply examples/text-generation/README.md]: @regisss fixed it now

File: examples/text-generation/README.md [hunk 2]
```diff
@@ -377,7 +372,6 @@ QUANT_CONFIG=./quantization_config/maxabs_measure.json python run_lm_eval.py \
 --trim_logits \
 --reuse_cache \
 --bf16
-```
```
Comment (regisss) [inline examples/text-generation/README.md:380]: This shouldn't be removed

File: examples/text-generation/README.md [hunk 3]
```diff
@@ -389,12 +383,7 @@ QUANT_CONFIG=./quantization_config/maxabs_quant_phi.json python run_generation.p
 --batch_size 1 \
 --bf16 \
 --trim_logits \
---reuse_cache \
---fp8
-```
```
Comment (regisss) [inline examples/text-generation/README.md:394]: Please keep the "```" after `--reuse_cache`

File: examples/text-generation/README.md [hunk 4]
```diff
@@ -284,7 +283,6 @@ QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
 --reuse_cache \
 --bf16 \
 --batch_size 1 \
```
Comment (regisss) [inline examples/text-generation/README.md:285]:
```suggestion
--batch_size 1
```

File: examples/text-generation/README.md [hunk 5]
```diff
@@ -302,7 +300,6 @@ QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
 --max_new_tokens 2048 \
 --max_input_tokens 2048 \
 --limit_hpu_graphs \
```
Comment (regisss) [inline examples/text-generation/README.md:302]:
```suggestion
--limit_hpu_graphs
```

File: examples/text-generation/README.md [hunk 6]
```diff
@@ -329,7 +326,6 @@ QUANT_CONFIG=./quantization_config/maxabs_quant_mixtral.json python run_generati
 --max_new_tokens 2048 \
 --batch_size 16 \
 --bf16 \
```
Comment (regisss) [inline examples/text-generation/README.md:328]:
```suggestion
--bf16
```

File: examples/text-generation/README.md [hunk 7]
```diff
@@ -361,7 +357,6 @@ QUANT_CONFIG=./quantization_config/maxabs_quant.json python ../gaudi_spawn.py \
...
 --reuse_cache \
 --trim_logits \
```
Comment (regisss) [inline examples/text-generation/README.md:359]:
```suggestion
--trim_logits
```

**Comment by Yantom1** [comment]
@regisss fixed the last comments also

**Review by regisss — APPROVED** [review]

**Pull request merged**